src/modules/user/user.controller.ts

```diff
@@ -0,0 +1,82 @@
+import { Controller, Post, Get, Put, Delete, Body, Headers, Res, HttpStatus, UsePipes } from '@nestjs/common';
+import { Response } from 'express';
+
+import { SchemaUserBuilder } from '../../validation/schema-builder/schema-user-builder';
+import { UserDto } from './dto/user.dto';
+import { RequestValidatorPipe } from 'src/pipes/request-validator.pipe';
+import { ResponseApiSuccess } from 'src/common/response-api';
+import { UserService } from './services/user.service';
+import { Users } from './models/users';
+import { editUserType } from './types';
+
+const createUserSchema = new SchemaUserBuilder().createUser<Body>().options({
+  abortEarly: false,
+  allowUnknown: true
+});
+
+@Controller('user')
+export class UserController {
+  constructor(public userService: UserService) {}
+  
+  @Post()
+  @UsePipes(new RequestValidatorPipe<UserDto>(createUserSchema))
+  createUser(@Res() response: Response, @Body() userDto: UserDto): void {
+    this.userService.createUser(userDto)
+    .then((result: Users) => {
```

Comment:
I recommend to use async-await instead of .then API for promises

Reply:
Done.